thangs_fetcher.py

```diff
@@ -685,22 +701,46 @@ def get_stl_search(self, stl_path):
         s = requests.Session()
 
         retries = Retry(total=10,
-                        backoff_factor=3,
+                        backoff_factor=1,
                         status_forcelist=[500, 502, 503, 504, 521],
-                        allowed_methods=frozenset(['GET', 'POST']),)
+                        allowed_methods=frozenset(['GET', 'POST', 'PUT']),)
         s.mount('https://', HTTPAdapter(max_retries=retries))
-        response = s.post(url, headers=headers, data=data)
-        response.raise_for_status()
+
+        try:
+            s.put(url=signedUrl, data=data) # params={'data': data}, args=(), 
+            #response = s.post(url, headers=headers, data=data)
+        except:
+            print("API Failed")
+            self.selectionSearching = False
+            self.searching = False
+            self.newSearch = False
+            self.SelectionFailed = True
+            return
 
         print("Select Search Returned")
 
+        try:
+            url_filepath = urllib.parse.quote(new_Filename, safe='')
+            url_filepath = re.sub(r'-', '%2D', url_filepath)

```

Comment:
Curious -- why are we replacing `-`?  IIRC it's valid in a url.

Reply:
Colin had said "The slashes and hyphens are most likely whats causing the issue", but he's looking into them right now I believe. Anything in the STL search isn't finished, just a placeholder.